social-circus/src/redux/actions/postActions.jsx

```diff
@@ -103,3 +102,36 @@ export const DislikePost = async (postID, token, user, dispatch) => {
     console.log(error.message);
   }
 };
+
+export const PostComment = async (postID, comment, dispatch) => {
+  dispatch(postLoading(true));
+  try {
+    const postRef = doc(db, "posts", postID);
+    await updateDoc(postRef, {
+      comments: arrayUnion(comment),
+    });
+    dispatch(getExplorePosts());
+  } catch (error) {
+    toast.error("Error : Couldn't post your comment.");
+    console.log(error);
+  } finally {
+    dispatch(postLoading(false));
+  }
+};
+
+export const DeleteComment = async (postID, comment, dispatch) => {
+  console.log(comment.commentID);
+  dispatch(postLoading(true));
+  try {
+    const postRef = doc(db, "posts", postID);
+    await updateDoc(postRef, {
+      comments: arrayRemove(comment),
+    });
+    dispatch(getExplorePosts());
```

Comment:
instead of entirely calling the `explorePosts` you could just update the comments array of your post. It would be performant

Reply:
That's an interesting approach Sharath. Will look into during refactor.